File: frontend/src/types/lead.ts
// Lead Types based on backend schemas

export interface LeadBase {
  campaign_id: string;
  first_name?: string;
  last_name?: string;
  email?: string;
  phone?: string;
  company?: string;
  title?: string;
  linkedin_url?: string;
  source_url?: string;
  raw_data?: Record<string, any>;
  email_verification?: Record<string, any>;
  enrichment_results?: Record<string, any>;
  enrichment_job_id?: string;
  email_copy_gen_results?: Record<string, any>;
  instantly_lead_record?: Record<string, any>;
}

export interface LeadCreate extends LeadBase {}

export interface LeadUpdate {
  first_name?: string;
  last_name?: string;
  email?: string;
  phone?: string;
  company?: string;
  title?: string;
  linkedin_url?: string;
  source_url?: string;
  raw_data?: Record<string, any>;
  email_verification?: Record<string, any>;
  enrichment_results?: Record<string, any>;
  enrichment_job_id?: string;
  email_copy_gen_results?: Record<string, any>;
  instantly_lead_record?: Record<string, any>;
}

export interface LeadResponse extends LeadBase {
  id: string;
  created_at: string;
  updated_at: string;
}

// Filter interfaces for API queries
export interface LeadListFilters {
  skip?: number;
  limit?: number;
  campaign_id?: string;
  email?: string;
  company?: string;
}

// API Response interfaces
export interface LeadListResponse {
  leads: LeadResponse[];
  total?: number;
}

// Search and filtering options
export interface LeadSearchFilters extends LeadListFilters {
  first_name?: string;
  last_name?: string;
  title?: string;
  has_email?: boolean;
  has_verification?: boolean;
  has_enrichment?: boolean;
} 

File: app/schemas/lead.py
from datetime import datetime
from typing import Optional, Any, Dict
from pydantic import BaseModel, Field

class LeadBase(BaseModel):
    campaign_id: str = Field(..., max_length=36, description="Campaign ID")
    first_name: Optional[str] = Field(None, max_length=100, description="First name")
    last_name: Optional[str] = Field(None, max_length=100, description="Last name")
    email: Optional[str] = Field(None, max_length=255, description="Email address")
    phone: Optional[str] = Field(None, max_length=50, description="Phone number")
    company: Optional[str] = Field(None, max_length=255, description="Company name")
    title: Optional[str] = Field(None, max_length=255, description="Job title")
    linkedin_url: Optional[str] = Field(None, max_length=255, description="LinkedIn URL")
    source_url: Optional[str] = Field(None, max_length=255, description="Source URL")
    raw_data: Optional[Dict[str, Any]] = Field(None, description="Raw data as JSON")
    email_verification: Optional[Dict[str, Any]] = Field(None, description="Email verification JSON")
    enrichment_results: Optional[Dict[str, Any]] = Field(None, description="Enrichment results JSON")
    enrichment_job_id: Optional[str] = Field(None, max_length=36, description="Enrichment job ID")
    email_copy_gen_results: Optional[Dict[str, Any]] = Field(None, description="Email copy gen results JSON")
    instantly_lead_record: Optional[Dict[str, Any]] = Field(None, description="Instantly lead record JSON")

class LeadCreate(LeadBase):
    pass

class LeadUpdate(BaseModel):
    first_name: Optional[str] = Field(None, max_length=100, description="First name")
    last_name: Optional[str] = Field(None, max_length=100, description="Last name")
    email: Optional[str] = Field(None, max_length=255, description="Email address")
    phone: Optional[str] = Field(None, max_length=50, description="Phone number")
    company: Optional[str] = Field(None, max_length=255, description="Company name")
    title: Optional[str] = Field(None, max_length=255, description="Job title")
    linkedin_url: Optional[str] = Field(None, max_length=255, description="LinkedIn URL")
    source_url: Optional[str] = Field(None, max_length=255, description="Source URL")
    raw_data: Optional[Dict[str, Any]] = Field(None, description="Raw data as JSON")
    email_verification: Optional[Dict[str, Any]] = Field(None, description="Email verification JSON")
    enrichment_results: Optional[Dict[str, Any]] = Field(None, description="Enrichment results JSON")
    enrichment_job_id: Optional[str] = Field(None, max_length=36, description="Enrichment job ID")
    email_copy_gen_results: Optional[Dict[str, Any]] = Field(None, description="Email copy gen results JSON")
    instantly_lead_record: Optional[Dict[str, Any]] = Field(None, description="Instantly lead record JSON")

class LeadResponse(LeadBase):
    id: str = Field(..., description="Lead ID")
    created_at: datetime = Field(..., description="Creation timestamp")
    updated_at: datetime = Field(..., description="Last update timestamp")

    class Config:
        from_attributes = True 